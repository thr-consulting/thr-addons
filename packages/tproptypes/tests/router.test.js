import React from 'react';
import {shallow} from 'enzyme';
import TPropTypes from '../src/index';

describe('Router Location', () => {
	const Comp = props => (
		<div>
			{props.location.hash}
			{props.location.key}
			{props.location.pathname}
			{props.location.search}
		</div>
	);
	Comp.propTypes = {
		location: TPropTypes.location.isRequired,
	};

	it('renders without error', () => {
		expect(shallow(
			<Comp
				location={{
					hash: '#myhash',
					pathname: '/this/is/my/path',
					search: '?var=123',
					state: {},
				}}
			/>,
		)).toMatchSnapshot();
	});
});

describe('RouterHistory', () => {
	const Comp = props => (
		<div>
			{props.history.action}
			{props.history.length}
		</div>
	);
	Comp.propTypes = {
		history: TPropTypes.history.isRequired,
	};

	it('renders without error', () => {
		expect(shallow(
			<Comp
				history={{
					action: 'PUSH',
					length: 15,
				}}
			/>,
		)).toMatchSnapshot();
	});
});
