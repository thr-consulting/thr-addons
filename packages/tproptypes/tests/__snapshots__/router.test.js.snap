// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router Location renders without error 1`] = `
<div>
  #myhash
  /this/is/my/path
  ?var=123
</div>
`;

exports[`RouterHistory renders without error 1`] = `
<div>
  PUSH
  15
</div>
`;
